### ISSUE TITLE
Class statics: Arrow functions not supported in goog.defineClass.

### ISSUE THREAD

**Issue by MarbleDaemon** [google/closure-compiler#1204]
The class was defined as followed:

```javascript
goog.provide('test.Test');

test.Test = good.defineClass(null, {
  constructor: function(){},

  statics: {
    test: () => console.log('This is a test');
  }
});
```

When compile, it throws a compiling error stating that arrow functions is not supported in goog.defineClass.

Previous version of compiler worked well with static arrow functions. Please explain or fix it. Thanks!

**Comment by MatrixFrog** [comment]
I don't recall when this changed but I suspect it's because the `this` inside such a function would not refer to what you'd expect. If you're using ES6, why not switch to `class`es?

**Comment by MarbleDaemon** [comment]
Hello,

As they are just static functions, there won't be any use of ```this```.

Converting all those classes to ES6 classes would take a lot of time and efforts.

**Comment by MatrixFrog** [comment]
Right, but statics in general can refer to `this` and it's a reference to the class itself. I think that's why we added the restriction. Since we don't have an automatic goog.defineClass->class converter I would recommend this syntax:

    statics: {
      test() { console.log('This is a test'); }
    }

which hopefully isn't *too* tedious to switch to.

**Comment by ChadKillingsworth** [comment]
As with #1160, we need a way to differentiate arrow functions that depend on `this` bindings. Once we have that, we won't have to treat all arrow functions the same. In fact, we might be able to transpile them differently too - bind is kinda slow.

**Comment by MatrixFrog** [comment]
I don't think we actually use bind when transpiling but anyway, yes I think that would be a fine way to fix this.

**Comment by ChadKillingsworth** [comment]
@MatrixFrog Your right - we use a closure.
